sycl/test/basic_tests/macros.cpp

```diff
@@ -1,18 +1,14 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
-//==------------------- macros.cpp - SYCL buffer basic test ----------------==//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-device-only -dM -E %s -o %t.device
+// RUN: %clangxx %fsycl-host-only -dM -E %s -o %t.host
 //
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// RUN: FileCheck %s < %t.device --check-prefixes=DEVICE,COMMON \
+// RUN:     --implicit-check-not=__SPIRV
+// RUN: FileCheck %s < %t.host --check-prefixes=COMMON \
+// RUN:      --implicit-check-not=__SPIRV
 //
-//===----------------------------------------------------------------------===//
+// COMMON-DAG: #define SYCL_LANGUAGE_VERSION
```

Comment:
Should it be checking for `__SYCL_COMPILER_VERSION` as well?

Reply:
Oh, please disregard my previous comment, it looks like I might have had outdated headers in my build directory: after doing `ninja sycl` I see many more defines - I will update the test further based on that list

Reply:
Restored the check in 3d1b313fabbd8dabf70460b65102ed3fb22afcce